### ISSUE TITLE
Tooltip example is invalid HTML and is missing aria-labelling

### ISSUE THREAD

**Issue by lee-chase** [carbon-design-system/carbon#771]
## Detailed description

The tooltip example shows an anchor with an active element (tabindex="0" inside it. This is not permitted by the spec.

- “there must be no interactive content or a element descendants”
  -  https://www.w3.org/TR/html5/single-page.html#elementdef-a 
- “Interactive content is content that is specifically intended for user interaction. […] The tabindex attribute can also make any element into interactive content.”
  - https://www.w3.org/TR/html5/single-page.html#interactive-content-2

In addition, there is no aria labelling see https://www.w3.org/TR/wai-aria-practices/#tooltip

> Is this issue related to a specific component? 
Tooltip

> What did you expect to happen? What happened instead? What would you like to see changed?
The example should be valid HTML, have role and aria-describedby attributes.

> What browser are you working in? 

> What version of the Carbon Design System are you using? 

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

## Additional information

* Screenshots or code
* Notes

## Add labels
Please choose the appropriate label(s) from our existing label list to ensure that your issue is properly categorized. This will help us to better understand and address your issue.

**Comment by alisonjoseph** [comment]
Tagging @aledavila since she is working on updates to tooltip that will be released in `v9` at the end of the month.

**Comment by alisonjoseph** [comment]
Ref PR https://github.com/carbon-design-system/carbon-components/pull/714
